scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/client/SchedulerNodeClient.java

```diff
@@ -0,0 +1,531 @@
+ * or a different license than the AGPL.
+ *
+ *  Initial developer(s):               The ProActive Team
+ *                        http://proactive.inria.fr/team_members.htm
+ *  Contributor(s):
+ *
+ *  * $$ACTIVEEON_INITIAL_DEV$$
+ */
+
+package org.ow2.proactive.scheduler.task.client;
+
+import org.apache.log4j.Logger;
+import org.objectweb.proactive.annotation.PublicAPI;
+import org.ow2.proactive.authentication.ConnectionInfo;
+import org.ow2.proactive.authentication.crypto.CredData;
+import org.ow2.proactive.db.SortParameter;
+import org.ow2.proactive.scheduler.common.*;
+import org.ow2.proactive.scheduler.common.exception.*;
+import org.ow2.proactive.scheduler.common.job.Job;
+import org.ow2.proactive.scheduler.common.job.JobId;
+import org.ow2.proactive.scheduler.common.job.JobInfo;
+import org.ow2.proactive.scheduler.common.job.JobPriority;
+import org.ow2.proactive.scheduler.common.job.JobResult;
+import org.ow2.proactive.scheduler.common.job.JobState;
+import org.ow2.proactive.scheduler.common.task.TaskId;
+import org.ow2.proactive.scheduler.common.task.TaskResult;
+import org.ow2.proactive.scheduler.common.task.TaskState;
+import org.ow2.proactive.scheduler.common.usage.JobUsage;
+import org.ow2.proactive.scheduler.common.util.logforwarder.AppenderProvider;
+import org.ow2.proactive.scheduler.job.SchedulerUserInfo;
+import org.ow2.proactive.scheduler.rest.ISchedulerClient;
+import org.ow2.proactive.scheduler.rest.SchedulerClient;
+import org.ow2.proactive.scheduler.task.utils.Decrypter;
+
+import java.security.KeyException;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * Scheduler api available as a variable during the execution of a task. it is based on the ISchedulerClient interface.
+ *
+ * @author The ProActive Team
+ */
+@PublicAPI
+public class SchedulerNodeClient implements ISchedulerClient {
+
+    private static final Logger logger = Logger.getLogger(SchedulerNodeClient.class);
+
+    private Decrypter decrypter;
+
+    private String url;
+
+    private ISchedulerClient client;
+
+    public SchedulerNodeClient(Decrypter decrypter, String url) {
+        SchedulerNodeClient.this.decrypter = decrypter;
+        SchedulerNodeClient.this.url = url;
+        client = SchedulerClient.createInstance();
+
+    }
+
+    /**
+     * Connects to the scheduler at the default url, using the current user credentials
+     *
+     * @throws Exception
+     */
+    public void connect() throws Exception {
+        connect(SchedulerNodeClient.this.url);
+    }
+
+    /**
+     * Connects to the scheduler at the specified url, using the current user credentials
+     * @param url url of the scheduler
+     * @throws Exception
+     */
+    public void connect(String url) throws Exception {
+        System.out.println("[SchedulerNodeClient] Connecting to scheduler at url " + url);
```

Comment:
This is printed inside the task execution, there are no loggers available, and this information is important for the user as he needs to know when we are trying to connect.